plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -362,17 +378,80 @@ func (m *MQTTConsumer) createOpts() (*mqtt.ClientOptions, error) {
 				server = "ssl://" + server
 			}
 		}
-
 		opts.AddBroker(server)
 	}
 	opts.SetAutoReconnect(false)
 	opts.SetKeepAlive(time.Second * 60)
 	opts.SetCleanSession(!m.PersistentSession)
 	opts.SetConnectionLostHandler(m.onConnectionLost)
-
 	return opts, nil
 }
 
+// parseMeasurement gets a single value from the incoming topic based on the user configuration (TopicParsing.Measurement)
+func parseMeasurement(keys []string, values []string) (string, error) {
+	for i, k := range keys {
+		if k != "_" {
+			return values[i], nil
+		}
+	}
```

Comment:
There should be only one entry for `measurement`, so it would probably be easier to have a specialized handling for the measurement. We should store the index and access the `values` directly.

Reply:
this has been implemented.